api/src/main/java/com/datastrato/gravitino/rel/TableCatalog.java

```diff
@@ -281,4 +283,9 @@ Table alterTable(NameIdentifier ident, TableChange... changes)
   default boolean purgeTable(NameIdentifier ident) throws UnsupportedOperationException {
     throw new UnsupportedOperationException("purgeTable not supported.");
   }
+
+  /** @return The set of capabilities that this catalog does not support. */
+  default Set<TableCatalogInability> inability() {
+    return Collections.emptySet();
+  }
```

Comment:
I suggest referencing the implementation of Trino connector capabilities  and using 'Capability' instead of 'Inability'.

Reply:
Using `Inability` instead of `Capability ` can provide users with more effective prompts for unsupported reasons.